src/text-buffer.coffee

```diff
@@ -1199,19 +1200,33 @@ class TextBuffer
   ###
 
   # Public: Save the buffer.
-  save: ->
-    @saveAs(@getPath())
+  save: (options) ->
+    @saveAs(@getPath(), options)
 
   # Public: Save the buffer at a specific path.
   #
   # * `filePath` The path to save at.
-  saveAs: (filePath) ->
+  saveAs: (filePath, options) ->
     unless filePath then throw new Error("Can't save buffer with no file path")
 
     @emitter.emit 'will-save', {path: filePath}
     @emit 'will-be-saved', this if Grim.includeDeprecatedAPIs
     @setPath(filePath)
-    @file.writeSync(@getText())
+
+    if options?.backup and @file.existsSync()
+      backupFilePath = filePath + '~'
+      backupFilePath += '~' while fs.existsSync(backupFilePath)
+      fs.writeFileSync(backupFilePath, @file.readSync())
+
+    try
+      @file.writeSync(@getText())
+      fs.removeSync(backupFilePath) if backupFilePath?
```

Comment:
Ah, this is kind of a bummer because `File.prototype.writeSync` has some conveniences related to privilege escalation... Can I just open a file descriptor for this path and `fdatasync` on it while allowing the write to proceed down the existing path, or do the write and the sync need to occur on the same file descriptor. I'm thinking they probably do, but thought I'd check before I go through the hassle.